source/json.c

```diff
@@ -291,6 +291,16 @@ bool aws_json_value_compare(const struct aws_json_value *a, const struct aws_jso
     return cJSON_Compare(cjson_a, cjson_b, is_case_sensitive);
 }
 
+struct aws_json_value *aws_json_value_duplicate(const struct aws_json_value *value) {
+    struct cJSON *cjson = (struct cJSON *)value;
+    if (cJSON_IsInvalid(cjson)) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        return NULL;
+    }
+
+    return (void *)cJSON_Duplicate(cjson, true);
```

Comment:
This function can also fail. I think we should do aws_raise_error if it fails.

Reply:
fair point. it returns null on failure. let me add a check for that